go/internal/flag/flag.go

```diff
@@ -0,0 +1,156 @@
+
+// Package flag is an internal package to allow us to gracefully transition
+// from the standard library's flag package to pflag. See VEP-4 for details.
+//
+// In general, this package should not be imported or depended on, except in the
+// cases of package servenv, and entrypoints in go/cmd. This package WILL be
+// deleted after the migration to pflag is completed, without any support for
+// compatibility.
+package flag
+
+import (
+	goflag "flag"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+
+	"vitess.io/vitess/go/vt/log"
+)
+
+// Parse wraps the standard library's flag.Parse to perform some sanity checking
+// and issue deprecation warnings in advance of our move to pflag.
+//
+// It also adjusts the global CommandLine's Usage func to print out flags with
+// double-dashes when a user requests the help, attempting to otherwise leave
+// the default Usage formatting unchanged.
+//
+// See VEP-4, phase 1 for details: https://github.com/vitessio/enhancements/blob/c766ea905e55409cddeb666d6073cd2ac4c9783e/veps/vep-4.md#phase-1-preparation
+func Parse() {
+	// First, override the Usage func to make flags show in their double-dash
+	// forms to the user.
+	goflag.CommandLine.Usage = func() {
+		fmt.Fprintf(goflag.CommandLine.Output(), "Usage of %s:\n", os.Args[0])
+
+		var buf strings.Builder
+		goflag.CommandLine.VisitAll(func(f *goflag.Flag) {
+			defer buf.Reset()
+			defer func() { fmt.Fprintf(goflag.CommandLine.Output(), "%s\n", buf.String()) }()
+
+			// See https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/flag/flag.go;l=512;drc=refs%2Ftags%2Fgo1.17.7
+			// for why two leading spaces.
+			buf.WriteString("  ")
+
+			// We use `UnquoteUsage` to preserve the "name override"
+			// behavior of the standard flag package, documented here:
+			//
+			//	> The listed type, here int, can be changed by placing a
+			//	> back-quoted name in the flag's usage string; the first
+			//	> such item in the message is taken to be a parameter name
+			//	> to show in the message and the back quotes are stripped
+			//	> from the message when displayed. For instance, given
+			//	>
+			//	>	flag.String("I", "", "search `directory` for include files")
+			//	>
+			//	> the output will be
+			//	>
+			// 	> 	-I directory
+			//	>		search directory for include files.
+			name, usage := goflag.UnquoteUsage(f)
+
+			// From the standard library documentation:
+			//	> For bool flags, the type is omitted and if the flag name is
+			//	> one byte the usage message appears on the same line.
+			if bf, ok := f.Value.(maybeBoolFlag); ok && bf.IsBoolFlag() && len(name) == 1 {
+				fmt.Fprintf(&buf, "-%s\t%s", f.Name, usage)
+				return
+			}
+
+			// First line: name, and, type or backticked name.
+			buf.WriteString("--")
+			buf.WriteString(f.Name)
+			if name != "" {
+				fmt.Fprintf(&buf, " %s", name)
+			}
+			buf.WriteString("\n\t")
+
+			// Second line: usage and optional default, if not the zero value
+			// for the type.
+			buf.WriteString(usage)
```

Comment:
If you, like me, were wondering, _but what if the usage is empty? does the standard library really still print a tab character?_ oh boy does it!!

<img width="505" alt="Screen Shot 2022-02-20 at 9 34 51 AM" src="https://user-images.githubusercontent.com/4276638/154847877-574cc260-63a3-4584-bfe0-169d5b7e00a9.png">

([playground](https://go.dev/play/p/rbjzkOukVbI))

Reply:
haha good catch